client/src/context/useKanbanContext.tsx

```diff
@@ -134,6 +127,61 @@ export const KanbanProvider: FunctionComponent = ({ children }): JSX.Element =>
     return cards;
   };
 
+  const moveCard = (destination: IColumn): void => {
+    if (!focusedCard) {
+      updateSnackBarMessage('No focus card found!', 'error');
+      return;
+    }
+
+    const dupBoard = Object.assign({}, activeBoard);
+    const columnsCopy: IColumn[] = cloneDeep(columns);
+
+    const colIndex = columns.findIndex((col) => col._id === focusedCard.columnId);
+
+    if (focusedCard.columnId !== destination._id) {
+      const targetColumnIndex = columnsCopy.findIndex((col) => col._id === destination._id);
+      if (targetColumnIndex > -1) {
+        const targetColumn = columnsCopy[targetColumnIndex];
+        const originalColumn = columnsCopy[colIndex];
+        columnsCopy[colIndex].cards = originalColumn.cards.splice(colIndex, 1);
+        focusedCard.columnId = targetColumn._id;
+        targetColumn.cards.push(focusedCard);
+      }
+    }
+
+    dupBoard.columns = columnsCopy;
+
+    updateBoard(dupBoard);
+    setActiveBoard(dupBoard);
+    setColumns(columnsCopy);
+  };
+
+  const copyCard = (destination: IColumn): void => {
+    if (!focusedCard) {
+      updateSnackBarMessage('No focus card found!', 'error');
+      return;
+    }
+    const source = getColumnById(focusedCard?.columnId as string);
+    const dupBoard = Object.assign({}, activeBoard);
+    const columnsCopy: IColumn[] = cloneDeep(columns);
+
+    if (source._id !== destination._id) {
+      const targetColumnIndex = columnsCopy.findIndex((col) => col._id === destination._id);
+      if (targetColumnIndex > -1) {
+        const targetColumn = columnsCopy[targetColumnIndex];
+        const card = focusedCard;
+        card._id = `item-${Math.floor(Math.random() * 999999)}`;
```

Comment:
Let's change this to uuid or nanoid for cards.

Reply:
Ok, I am using uuidv4 throughout useKanbanContext now. That was very easy to implement :)

Reply:
This will be very temporary, when we refactor to have Card models and Columns we will have to populate this with the ID sent back from the server. But this looks good for now!